src/response.rs

```diff
@@ -436,6 +436,16 @@ where
 
         Ok(())
     }
+
+    /// Retrieves the current value of the `Response` status code
+    pub fn get_status_code(&self) -> StatusCode {
+        self.status_code.clone()
+    }
+
+    /// Retrieves the current value of the `Response` data length
+    pub fn get_data_length(&self) -> Option<usize> {
+        self.data_length.clone()
```

Comment:
`.clone()` is superfluous here. `Option<usize>` is `Copy` https://doc.rust-lang.org/std/option/enum.Option.html#impl-Copy

Reply:
Ohh! Thanks for sharing! Fixing it right away!